### ISSUE TITLE
Private github repositories?

### ISSUE THREAD

**Issue by natdm** [zokugun/vscode-vsix-manager#13]
It doesn't look like this code is set to look at private github repos, no spot for an auth token or parsing of an environmental variable for it. Is there a way to do this?

**Comment by natdm** [comment]
Agreed, I've got them in env vars now, so a setting to point to an env var would be a good idea. That's probably a pretty common usecase.

**Comment by GitMensch** [comment]
@daiyam Any idea about an implementation as outlined?

**Comment by daiyam** [comment]
We will need to add a new source type (GitHub) to have something like
```
"vsix.sources": {
    "mgh": {
        "type": "github",
        "repository": "<username>",
        "token": "env:MY_TOKEN",
    },
},
"vsix.extensions": [
        "mgh:<project>",
    ],
```

**Comment by GitMensch** [comment]
Sounds good, I'd suggest to go with vscode's `${env:MY_TOKEN}` and support `${env:name}` also in the `repository`, `serviceUrl` and `path` properties

**Comment by daiyam** [comment]
Hi, it's late but the newest version (`0.7.0`) should support private repositories. Sorry for the long wait.
